openmc/plotter.py

```diff
@@ -0,0 +1,588 @@
+from numbers import Integral, Real
+
+import numpy as np
+
+import openmc.checkvalue as cv
+import openmc.data
+
+# Supported keywords for xs plotting
+PLOT_TYPES = ['total', 'scatter', 'elastic', 'inelastic', 'fission',
+              'absorption', 'capture', 'nu-fission', 'nu-scatter', 'unity',
+              'slowing-down power', 'damage']
+
+# Special MT values
+UNITY_MT = -1
+XI_MT = -2
+
+# MTs to combine to generate associated plot_types
+_INELASTIC = [mt for mt in openmc.data.SUM_RULES[3] if mt not in [5, 27]]
+PLOT_TYPES_MT = {'total': openmc.data.SUM_RULES[1],
+                 'scatter': [2] + _INELASTIC,
+                 'elastic': [2],
+                 'inelastic': _INELASTIC,
+                 'fission': [18],
+                 'absorption': [27], 'capture': [101],
+                 'nu-fission': [18],
+                 'nu-scatter': [2] + _INELASTIC,
+                 'unity': [UNITY_MT],
+                 'slowing-down power': [2] + _INELASTIC + [XI_MT],
+                 'damage': [444]}
+# Operations to use when combining MTs the first np.add is used in reference
+# to zero
+PLOT_TYPES_OP = {'total': (np.add,),
+                 'scatter': (np.add,) * (len(PLOT_TYPES_MT['scatter']) - 1),
+                 'elastic': (),
+                 'inelastic': (np.add,) * (len(PLOT_TYPES_MT['inelastic']) - 1),
+                 'fission': (), 'absorption': (),
+                 'capture': (), 'nu-fission': (),
+                 'nu-scatter': (np.add,) * (len(PLOT_TYPES_MT['nu-scatter']) - 1),
+                 'unity': (),
+                 'slowing-down power':
+                    (np.add,) * (len(PLOT_TYPES_MT['slowing-down power']) - 2) + (np.multiply,),
+                 'damage': ()}
+
+# Whether or not to multiply the reaction by the yield as well
+PLOT_TYPES_YIELD = {'total': (False, False),
+                    'scatter': (False,) * len(PLOT_TYPES_MT['scatter']),
+                    'elastic': (False,),
+                    'inelastic': (False,) * len(PLOT_TYPES_MT['inelastic']),
+                    'fission': (False,), 'absorption': (False,),
+                    'capture': (False,), 'nu-fission': (True,),
+                    'nu-scatter': (True,) * len(PLOT_TYPES_MT['nu-scatter']),
+                    'unity': (False,),
+                    'slowing-down power':
+                        (True,) * len(PLOT_TYPES_MT['slowing-down power']),
+                    'damage': (False,)}
```

Comment:
Oh that was because I thought I would be smarter than the code and explicitly give values of false or true for each reaction. Then when I saw that the runtime penalty wasn't so bad I didn't bother but left the generality in. Since its trivial to put it back in when needed I will generalize this.